lib/ui/screens/new_contact_screen.dart

```diff
@@ -0,0 +1,149 @@
+import 'package:flutter/material.dart';
+
+class NewContact extends StatefulWidget {
+    const NewContact({Key? key }) : super(key: key);
+
+    @override
+    _NewContactState createState() => _NewContactState();
+}
+
+class _NewContactState extends State<NewContact> {
+    late String firstName; 
+    late String lastName; 
+    late String phoneNumber; 
+    String gender = ""; 
+    late String email; 
+    late String address; 
+
+    List<String> genderList = ["Male", "Female", "Others"]; 
+
+    final GlobalKey<FormState> formKey = GlobalKey<FormState>(); 
+
+    Widget firstNameField(){
+        return TextFormField(
+            decoration: InputDecoration(labelText: "First name"),
+            validator: (String ? value) {
+                if (value==null || value.isEmpty){
+                    return "First name is required"; 
+                }
+            },
+            onSaved: (String ? value) => firstName = value!,
+        ); 
+    }
+
+    Widget lastNameField(){
+        return TextFormField(
+            decoration: InputDecoration(labelText: "Last name"),
+            validator: (String ? value) {
```

Comment:
Should create a Validator class, put it in /lib/services/validator.dart

```suggestion
            class Validator {
  static bool isEmail(String input) {
    /// Put validation here
  }

  static bool isPhoneNumber(String input) {
    /// Put validation here
  }
}

```